force-app/main/default/classes/HierarchySettingsService.cls

```diff
@@ -113,18 +122,59 @@ public virtual with sharing class HierarchySettingsService {
             }
         } catch (UTIL_Describe.SchemaDescribeException e) {
             UTIL_Debug.debug('****SchemaDescribeException encountered: ' + e.getMessage());
+            System.debug('****SchemaDescribeException encountered: ' + e.getMessage());
             throw new InvalidSettingsException(Label.stgErrorInvalidSettings);
         } catch (System.TypeException e) {
             UTIL_Debug.debug('****TypeException encountered: ' + e.getMessage());
+            System.debug('****TypeException encountered: ' + e.getMessage());
             throw new InvalidSettingsException(Label.stgErrorInvalidSettings);
         } catch (Exception e) {
             UTIL_Debug.debug('****Unexpected Exception encountered: ' + e.getMessage());
+            System.debug('****Unexpected Exception encountered: ' + e.getMessage());
             throw new InvalidSettingsException(Label.stgErrorInvalidSettings);
         }
 
         return this.locateHierarchySettingsMapper().updateHierarchySettings(hierarchySettingsModelToUpdate);
     }
 
+    /******************************************************************************************************************************
+     * @description Append the hed__ prefix to all hierarchy settings field names specified in the settingsSingleValueBySettingsName
+     * and settingsListSettingsName of a HierarchySettingsChangesVModel when handled in a managed context.
+     * @param  hierarchySettingsChangesVModelToNamespace A HierarchySettingsChangesVModel with fields requiring a namespace prefix
+     * @return A HierarchySettingsChangesVModel
+     ******************************************************************************************************************************/
+    private HierarchySettingsChangesVModel addNamespacePrefixToHierarchySettingsChangesFields(
```

Comment:
This method seems a little heavy, let alone by making extra loops for the task at hand. Also, if we are to perform something of this nature in this fashion, I suggest instantiating a new HierarchySettingsChangesVModel and return it that model to avoid side effect.